x-pack/plugins/security_solution/public/common/components/link_to/index.ts

```diff
@@ -39,20 +39,66 @@ export const useFormatUrl = (page: SecurityPageName) => {
 
   const formatUrl = useCallback<FormatUrl>(
     (path: string, { absolute = false, skipSearch = false } = {}) => {
-      const pathArr = path.split('?');
-      const formattedPath = `${pathArr[0]}${
-        !skipSearch
-          ? isEmpty(pathArr[1])
-            ? search
-            : `?${pathArr[1]}${isEmpty(search) ? '' : `&${search}`}`
-          : isEmpty(pathArr[1])
-          ? ''
-          : `?${pathArr[1]}`
-      }`;
+      const formattedPath = formatPath(path, search, skipSearch);
       return getAppUrl({ deepLinkId: page, path: formattedPath, absolute });
     },
     [getAppUrl, page, search]
   );
 
   return { formatUrl, search };
 };
+
+type GetSecuritySolutionUrl = (param: {
+  deepLinkId: SecurityPageName;
+  path?: string;
+  absolute?: boolean;
+  skipSearch?: boolean;
+}) => string;
+
+export const useGetSecuritySolutionUrl = () => {
+  const { getAppUrl } = useAppUrl();
+  const getUrlStateQueryString = useGetUrlStateQueryString();
+
+  const getSecuritySolutionUrl = useCallback<GetSecuritySolutionUrl>(
+    ({ deepLinkId, path = '', absolute = false, skipSearch = false }) => {
+      const search = needsUrlState(deepLinkId) ? getUrlStateQueryString() : '';
+      const formattedPath = formatPath(path, search, skipSearch);
+      return getAppUrl({ deepLinkId, path: formattedPath, absolute });
+    },
+    [getAppUrl, getUrlStateQueryString]
+  );
+
+  return getSecuritySolutionUrl;
+};
+
+function formatPath(path: string, search: string, skipSearch?: boolean) {
+  const [urlPath, parameterPath] = path.split('?');
+  const formattedPath = `${urlPath}${
+    !skipSearch
+      ? isEmpty(parameterPath)
+        ? search
+        : `?${parameterPath}${isEmpty(search) ? '' : `&${search}`}`
+      : isEmpty(parameterPath)
+      ? ''
+      : `?${parameterPath}`
+  }`;
+  return formattedPath;
+}
+
+// TODO use links.needsUrlState function instead when implemented and remove this function
```

Comment:
should be there now

Reply:
cool! I will integrate links config in the next PR